druid-shell/src/platform/x11/screen.rs

```diff
@@ -14,9 +14,136 @@
 
 //! X11 Monitors and Screen information.
 
+use x11rb::connection::Connection;
+use x11rb::errors::ReplyOrIdError;
+use x11rb::protocol::randr::{self, ConnectionExt as _, Crtc};
+use x11rb::protocol::xproto::{Screen, Timestamp};
+
+use crate::kurbo::Rect;
 use crate::screen::Monitor;
 
+fn monitor<Pos>(primary: bool, (x, y): (Pos, Pos), (width, height): (u16, u16)) -> Monitor
+where
+    Pos: Into<i32>,
+{
+    let rect = Rect::from_points(
+        (x.into() as f64, y.into() as f64),
+        (width as f64, height as f64),
+    );
+    // TODO: Support for work_rect. It's complicated...
+    Monitor::new(primary, rect, rect)
+}
+
 pub(crate) fn get_monitors() -> Vec<Monitor> {
-    tracing::warn!("Screen::get_monitors() is currently unimplemented for X11 platforms.");
-    Vec::new()
+    // TODO: Change API to pass in the App. Until then, this creates a separate X11 connection
+    let (conn, screen_num) = match x11rb::connect(None) {
+        Ok(res) => res,
+        Err(err) => {
+            tracing::error!("Error in Screen::get_monitors(): {:?}", err);
+            return Vec::new();
+        }
+    };
+    match get_monitors_impl(&conn, screen_num) {
```

Comment:
You can the get `Application` here. but it is better to create a connection if there is no global application.
```suggestion
    let result = if let Some(app) = crate::Application::try_global() {
        let app = app.platform_app;
        get_monitors_impl(app.connection().as_ref(), app.screen_num() as usize)
    } else {
        let (conn, screen_num) = match x11rb::connect(None) {
            Ok(res) => res,
            Err(err) => {
                tracing::error!("Error in Screen::get_monitors(): {:?}", err);
                return Vec::new();
            }
        };
        get_monitors_impl(&conn, screen_num)
    };
    match result {
```